tests/vehicle_test.cpp

```diff
@@ -266,3 +273,152 @@ TEST_CASE( "Unfolding vehicle parts and testing degradation", "[item][degradatio
 
     clear_vehicles( &get_map() );
 }
+
+// Basically a copy of vehicle::connect() that uses an arbitrary cord type
+static void connect_power_line( const tripoint &src_pos, const tripoint &dst_pos,
+                                const itype_id &itm )
+{
+    map &here = get_map();
+    item cord( itm );
+    cord.set_var( "source_x", src_pos.x );
+    cord.set_var( "source_y", src_pos.y );
+    cord.set_var( "source_z", src_pos.z );
+    cord.set_var( "state", "pay_out_cable" );
+    cord.active = true;
+
+    const optional_vpart_position target_vp = here.veh_at( dst_pos );
+    const optional_vpart_position source_vp = here.veh_at( src_pos );
+
+    if( !target_vp ) {
+        return;
+    }
+    vehicle *const target_veh = &target_vp->vehicle();
+    vehicle *const source_veh = &source_vp->vehicle();
+    if( source_veh == target_veh ) {
+        return ;
+    }
+
+    tripoint target_global = here.getabs( dst_pos );
+    const vpart_id vpid( cord.typeId().str() );
+
+    point vcoords = source_vp->mount();
+    vehicle_part source_part( vpid, "", vcoords, item( cord ) );
+    source_part.target.first = target_global;
+    source_part.target.second = target_veh->global_square_location().raw();
+    source_veh->install_part( vcoords, source_part );
+
+    vcoords = target_vp->mount();
+    vehicle_part target_part( vpid, "", vcoords, item( cord ) );
+    tripoint source_global( cord.get_var( "source_x", 0 ),
+                            cord.get_var( "source_y", 0 ),
+                            cord.get_var( "source_z", 0 ) );
+    target_part.target.first = source_global;
+    target_part.target.second = source_veh->global_square_location().raw();
+    target_veh->install_part( vcoords, target_part );
+}
+
+TEST_CASE( "power_cable_stretch_disconnect" )
+{
+    clear_map();
+    clear_avatar();
+    map &m = get_map();
+    m.load( { 0, 0, 0 }, false );
```

Comment:
The test should be generalized to work at any map coordinates (if it doesn't already). Or you can use `game::place_player_overmap()` if you absolutely need this anchored at zero.

The load on its own breaks some game-wide assumptions, but they may not have been valid in the first place (as seen in #60674).

Reply:
I see, cheers! I didn't think too much of it, but after looking at it further it did expose a bug. Should be fixed in #60714